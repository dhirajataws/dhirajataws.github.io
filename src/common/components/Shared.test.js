import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { LogoImg, addLogoName } from './Shared';
test('LogoImg components ', () => {
    const tree = renderer.create(<LogoImg />).toJSON();
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('align-self', 'center')
    expect(tree).toHaveStyleRule('z-index', '3')
    expect(tree).toHaveStyleRule('height', '80%')
    expect(tree).toHaveStyleRule('width', '15%')
})

test('Image Container components ', () => {
    const values = { xs: "6", md: "4", sm: "6", lg: "4" }
    const withLogo = addLogoName(values);
    const testComponent = () => <div>test</div>
    const tree = renderer.create(withLogo(testComponent))
    const testInstance = tree.root;
    expect(testInstance.findByType(testComponent));
    expect(testInstance.children.length).toEqual(1);
})

