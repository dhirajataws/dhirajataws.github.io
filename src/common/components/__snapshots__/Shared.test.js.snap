// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoImg components  1`] = `
.c0 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  width: 15%;
  height: 80%;
  z-index: 3;
}

<img
  className="c0"
/>
`;
